src/commands/undeclare.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+SHRBINDIR=$(dirname $BASH_SOURCE)
+source $SHRBINDIR/utils.sh
+
+function undeclare {
+  # Show already declared packages that will be removed
+  if (( $(filter_declaredpkgs $@ | wc -l) > 0 )); then
+    echo -e "Removing declarations:\n\t$(filter_declaredpkgs $@ | xargs)"
+    for pkg in $(filter_declaredpkgs $@); do
+      # delete line starting with package name (optionally followed by spaces and an optional comment)
+      ${SUDO} sed -i '' -E "/^${pkg}[[:blank:]]*(#.*)?$/d" "$KEEPLISTFILE"
```

Comment:
```suggestion
      ${SUDO} sed -i '' -E "/^$[ \t]*{pkg}[[:blank:]]*(#.*)?$/d" "$KEEPLISTFILE"
```

This pattern to find the package to undeclare would support leading whitespaces. This will make it consistent with the current parsing of the `packages.list` file in the parse_keeplist_file method, which supports leading whitespace.